lib/components/clipboard-button/index.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactDom from 'react-dom';
+import Clipboard from 'clipboard';
+
+function ClipboardButton({ text, disabled }) {
+  const buttonRef = React.useRef();
+
+  const textCallback = React.useRef();
+  const successCallback = React.useRef();
+
+  const onCopy = () => {
+    setCopied(true);
+  };
+
+  const [isCopied, setCopied] = React.useState(false);
+
+  // toggle the `isCopied` flag back to `false` after 4 seconds
+  React.useEffect(() => {
+    if (isCopied) {
+      const confirmationTimeout = setTimeout(() => setCopied(false), 4000);
+      return () => clearTimeout(confirmationTimeout);
+    }
+  }, [isCopied]);
+
+  // update the callbacks on rerenders that change `text` or `onCopy`
+  React.useEffect(() => {
+    textCallback.current = () => text;
+    successCallback.current = onCopy;
+  }, [text, onCopy]);
+
+  // create the `Clipboard` object on mount and destroy on unmount
+  React.useEffect(() => {
+    const buttonEl = ReactDom.findDOMNode(buttonRef.current);
```

Comment:
Maybe I'm misunderstanding but I don't think we should ever need `findDOMNode()` with refs - I noted we are using `useRef()` instead of `createRef()` and learned that `useRef()` exists and should be used in these situations so that it doesn't create a new `ref` on every render.

```js
// bail if we've unmounted or something by the time this gets called
if ( ! buttonRef.current ) {
  return;
}

const clipboard = new Clipboard( buttonRef.current, {
  …
} );

…
```